app/core-components/modals/ModalRoot.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import type { State } from 'types/state';
+import modals from 'modules/modals';
+import DeleteTopicModal from './DeleteTopicModal';
+
+const { getModal } = modals.selectors;
+
+type StateProps = {
+  id: string,
+  modalType: string,
+};
+
+type Props = StateProps;
+
+const MODAL_COMPONENTS = {
```

Comment:
For the moment, yes. Unless you can think of a use case where we need multiple.

Reply:
The definition of a modal dictates that only one instance can be unique at a time. I just want to make sure this is the case here.

About the mapping, can't you pass a modal component as a prop instead of a string?

Reply:
Apparently not that easy, so I'll leave it for now
`Element type is invalid: **expected a string** (for built-in components) or a class/function (for composite components) but got: undefined`